package proto_ff;

// calm 自定义的fieldoption都在该文件中
import "yd_fieldoptions.proto";
import "common_logic.proto";


//服务器和服务器之间的协议号定义
enum ServerInternalCmd
{
  //系统 1 - 300
  SYS_PING						= 1;			//ping消息
  SYS_CLOSE_SERVER				= 2;			//关闭服务器

  SYS_REDIRECT_CLIENT_BY_GATE		= 10;			//通过网关重定向到客户端
  SYS_REDIRECT_CLIENT_BY_CENTER	= 11;			//通过中心服重定向到客户端
  SYS_REDIRECT_SOCIAL_BY_CENTER	= 12;			//通过中心服重定向到社交服
  SYS_REDIRECT_LOGIC_BY_CENTER    = 13;			//通过中心服重定向到逻辑服
  SYS_REDIRECT_LOGIC_ID_BY_CENTER	= 14;			//通过中心服重定向到指定ID的逻辑服
  SYS_REDIRECT_SERVER_BY_GATE		= 15;			//通过网关重定向到其他服务器


  LOGIC_REGISTER_CENTER            = 16;			// 逻辑服向中心服注册
  LOGIC_REGISTER_CENTER_RSP		 = 17;			// 中心服返回逻辑服 注册结果

  LOGIC_REGISTER_GATE              = 18;			// 逻辑服向网关注册
  LOGIC_REGISTER_GATE_RSP			 = 19;			// 网关返回逻辑服注册 结果

  CENTER_REGISTER_GATE			 = 20;			// 中心服向网关注册
  CENTER_REGISTER_GATE_RSP		 = 21;			// 网关返回中心服 注册结果

  CENTER_REGISTER_DB				 = 22;			// 中心服向DB服注册

  SOCIAL_REGISTER_CENTER			 = 23;			// 社交服向中心服注册
  SOCIAL_REGISTER_CENTER_RSP		 = 24;			// 中心服返回社交服 注册结果

  SOCIAL_QUERY_CENTER_INFO		 = 25;			// 社交服查询中心服信息(开服时间，是否加载完成，是否合服)
  SOCIAL_QUERY_CENTER_INFO_RSP	 = 26;			// 中心服返回社交服 服务器信息

  NOTIFY_LOGIC_STATE				 = 27;			// 中心服通知社交服，各个逻辑服的状态

  INNER_LOG_REQ					 = 31;			//内部日志请求

  NOTIFY_LOGIC_DATA_UPDATE		 = 32;			//通知逻辑服 数据更新(0点更新)

  //登录 300 - 350
  DB_ROLE_LIST_REQ				 = 301;			//从DB加载角色列表
  DB_ROLE_LIST_RSP				 = 302;			//DB返回角色列表
  DB_CREATE_ROLE_REQ				 = 303;			//创角请求
  DB_CREATE_ROLE_RSP				 = 304;			//返回创角结果
  DB_ROLE_INFO_REQ				 = 305;			//角色信息请求
  DB_ROLE_INFO_RSP				 = 306;			//返回角色信息
  DB_ROLE_INFO_UPDAT_REQ			 = 307;			//更新角色信息

  NOTIFY_LOGIC_ENTER_GAME_REQ		 = 308;			//中心服通知逻辑服进入游戏
  NOTIFY_LOGIC_ENTER_GAME_RSP		 = 309;			//逻辑服返回中心服进入游戏结果
  NOTIFY_LOGIC_LEAVE_GAME_REQ		 = 310;			//中心服通知逻辑服退出游戏
  NOTIFY_LOGIC_LEAVE_GAME_RSP		 = 311;			//逻辑服返回中心服退出游戏结果
  NOTIFY_CENTER_KICK_REQ			 = 312;			//逻辑服通知中心服踢人
  NOTIFY_GATE_ENTER_GAME			 = 313;			//中心服通知网关进入游戏
  NOTIFY_GATE_CHANGE_LOGIC		 = 314;			//中心服通知网关切换逻辑服
  NOTIFY_GATE_LEAVE_GAME			 = 315;			//中心服通知网关退出游戏
  NOTIFY_GATE_RECONNECT_GAME		 = 316;			//中心服通知网关重连游戏
  NOTIFY_CENTER_DISCONNECT		 = 317;			//网关通知中心服客户端掉线
  NOTIFY_LOGIC_DISCONNECT			 = 318;			//中心服通知逻辑服断开连接
  NOTIFY_LOGIC_RECONNECT			 = 319;			//中心服通知逻辑服 重连

};